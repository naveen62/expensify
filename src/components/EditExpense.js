import React from  'react'
import {connect} from 'react-redux';
import ExpenseForm from './ExpenseForm';
import {editExpense, removeExpense} from '../actions/expense'

export class EditExpense extends React.Component {
    onSubmit = (expense) => {
        this.props.editExpense(this.props.match.params.id, expense)
        this.props.history.push('/')
    }
    handleRemove = () => {
        this.props.removeExpense(this.props.match.params.id)
        this.props.history.push('/')
    }

    render() {
        return (
            <div>
                <ExpenseForm
                 expense={this.props.expense}
                 onSubmit={this.onSubmit} />
                    <button onClick={this.handleRemove} >Remove</button>       
            </div>
        ) 
    }
}

const mapStateToProps = (state, props) => {
    return {
        expense: state.expenses.find((expense) => {
            return expense.id === props.match.params.id
        })
    }
}
const mapDispatchToProps = (dispatch) => {
    return {
        editExpense: (id, expense) => dispatch(editExpense(id, expense)),
        removeExpense: (id) => dispatch(removeExpense({id}))
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(EditExpense);
