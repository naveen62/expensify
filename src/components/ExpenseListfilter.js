import React from 'react';
import {connect} from 'react-redux';
import {setStartDate, setEndDate ,setTextFilter, sortByAmount, sortByDate } from "../actions/filter";
import {DateRangePicker} from 'react-dates'

export class ExpenseListfilter extends React.Component {
   state = {
       calendarFocused: null
   }
   onDatesChange = ({startDate, endDate}  ) => {
        this.props.setStartDate(startDate);
        this.props.setEndDate(endDate);
   }
   onFocusChange = (cal) => {
       this.setState(() => ({calendarFocused: cal}))
   }    
   onTextChange = (e) => {
    this.props.setTextFilter(e.target.value)
   }
   onSortChange = (e) => {
    if(e.target.value === 'date') {
        this.props.sortByDate()
    } else if(e.target.value === 'amount') {
        this.props.sortByAmount()
    }
    }
    render() {
        return (
            <div>
                <input type='text' value={this.props.filter.text} onChange={this.onTextChange} />
                <select onChange={this.onSortChange}>
                    <option value='date'>Date</option>
                    <option value='amount'>Amount</option>
                </select>
                <DateRangePicker
                    startDate={this.props.filter.startDate}
                    endDate={this.props.filter.endDate}
                    onDatesChange={this.onDatesChange}
                    focusedInput={this.state.calendarFocused}
                    onFocusChange={this.onFocusChange}
                    numberOfMonths={1}
                    showClearDates={true}
                    isOutsideRange={() => false}
                 /> 
            </div>
        )
    }
}



const mapsStateToProps = (state) => {
    return {
        filter: state.filter
    }
}
const mapDispatchToProps = () => ({
    setTextFilter: (text) => dispatch(setTextFilter(text)),
    sortByDate: () => dispatch(sortByDate()),
    sortByAmount: () => dispatch(sortByAmount()),
    setStartDate: (startDate) => dispatch(setStartDate(startDate)),
    setEndDate: (endDate) => dispatch(setEndDate(endDate))
})
export default connect(mapsStateToProps, mapDispatchToProps)(ExpenseListfilter)