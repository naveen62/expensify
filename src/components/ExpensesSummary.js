import React from 'react';
import {connect} from 'react-redux';
import selectExpenses from "../selectors/expenses";
import getExpenseTotal from '../selectors/selectExpenseTotal';
import numeral from 'numeral'

class ExpensesSummary extends React.Component {

    render() {
        return (
            <div>
                <h3> Viewing {this.props.expenses.length} {this.props.expenses.length > 1 ? 'expenses' : 'expense' } totalling  {numeral(this.props.totalExpenses / 100).format('$0,0.00')} </h3>
            </div>
        )
    }
}
const mapStateToProps = (state) => {
    return {
        expenses: selectExpenses(state.expenses, state.filter),
        totalExpenses: getExpenseTotal(state.expenses)
    }
}
export default connect(mapStateToProps)(ExpensesSummary);