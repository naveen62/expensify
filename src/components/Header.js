import React from 'react';
import {NavLink} from  'react-router-dom';

const Header = () => {
    return (
        <div>
        <header>
            <h1>Expensify</h1>
            <NavLink activeClassName="is-active" exact={true} to="/">Home</NavLink>
            <NavLink activeClassName="is-active" to="/create">Create</NavLink>
        </header>
        </div>
    )
}
export default Header;