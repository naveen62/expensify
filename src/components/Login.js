import React from 'react';
import {connect} from 'react-redux';
import {startLogin} from '../actions/auth';

const Login = (props) => {
  return(
    <div>
        <button onClick={() => {
          props.dispatch(startLogin()).then(() => {
            console.log('login succsfull')
          })
        }} >Login</button>
    </div> 
  )  
}
export default connect()(Login)

