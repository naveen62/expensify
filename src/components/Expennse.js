import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListfilter  from './ExpenseListfilter';

export const Expense = () => (
    <div>
        <ExpenseListfilter />
        <ExpenseList />
    </div>
)
export default Expense;