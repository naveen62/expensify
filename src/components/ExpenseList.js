import React from 'react';
import {connect} from 'react-redux';
import ExpenseListitems from './ExpenseListitems';
import selectExpenses from '../selectors/expenses';

export const ExpenseList = (props) => (
    <div>
        {props.expenses.length === 0 ? (
            <p>No Expenses</p>
        ) : (
            props.expenses.map((expense) => <ExpenseListitems key={expense.id} {...expense} /> )
        ) }
        
    </div>
)

const mapStateToProps = (state) => {
    return {
        expenses: selectExpenses(state.expenses, state.filter)
    }
} 
export default connect(mapStateToProps)(ExpenseList);
