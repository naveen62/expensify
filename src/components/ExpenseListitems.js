import React from 'react';
import {connect} from 'react-redux';
import {removeExpense} from '../actions/expense';
import ExpenseList from './ExpenseList';
import { NavLink } from 'react-router-dom';

// const ExpenseListitems = ({description, amount, createdAt}) => (
//     <div>
//         <h1>{description}</h1>
//         <p>{amount} - {createdAt}</p>
//         <button>Remove</button>
//     </div>
// )
class ExpenseListitems extends React.Component {
    constructor(props) {
        super(props)
    }
    render() {
        return (
            <div>
            <NavLink to={`/edit/${this.props.id}`} exact={true}>
                <h1>{this.props.description}</h1>
            </NavLink>
                <p>{this.props.amount} - {this.props.createdAt}</p>
            </div>
        )
    }
}
export {ExpenseListitems};
export default connect()(ExpenseListitems)