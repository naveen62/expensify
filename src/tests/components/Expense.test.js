import React from 'react';
import {shallow} from 'enzyme';
import {Expense} from '../../components/Expennse';

test('should render Expense dashboard', () => {
    const wrapper = shallow(<Expense />);
    expect(wrapper).toMatchSnapshot()
})