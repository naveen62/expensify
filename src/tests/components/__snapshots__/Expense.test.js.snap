// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Expense dashboard 1`] = `
<div>
  <Connect(ExpenseListfilter) />
  <Connect(ExpenseList) />
</div>
`;
